testkit/src/test/java/org/apache/calcite/test/DiffRepositoryTest.java

```diff
@@ -57,4 +57,21 @@ public class DiffRepositoryTest {
     }
     assertThat("First assertion must always fail", assertPassed, is(false));
   }
+
+  @Test void testOutOfOrderTestCases() throws Exception {
+    DiffRepository r = DiffRepository.lookup(DiffRepositoryTestOutOfOrder.class);
+
+    IllegalArgumentException thrown = null;
+    try {
+      r.checkActualAndReferenceFiles();
+    } catch (IllegalArgumentException e) {
+      thrown = e;
+    }
+
+    // We expect an error about out-of-order test cases
+    assertThat("checkActualAndReferenceFiles should throw an error for out-of-order tests",
+        thrown != null, is(true));
+    assertThat("Error message should mention out-of-order or alphabetical",
```

Comment:
If we get a specific "out-of-order" error when calling `DiffRepository#assertEquals` then we can keep the existing generic error message in `checkActualAndReferenceFiles`. In most cases, we are hitting the code of `DiffRepository#getTestCaseElement` first before we reach `checkActualAndReferenceFiles`.